packages/mask/src/components/shared/MaskApplicationBox.tsx

```diff
@@ -0,0 +1,324 @@
+        top: 0,
+        width: '100%',
+        zIndex: 2,
+        paddingTop: theme.spacing(1),
+        paddingBottom: theme.spacing(2),
+        backgroundColor: theme.palette.background.paper,
+    },
+    tab: {
+        height: 36,
+        minHeight: 36,
+        fontWeight: 300,
+    },
+    tabs: {
+        width: 552,
+        height: 36,
+        minHeight: 36,
+        margin: '0 auto',
+        borderRadius: 4,
+        backgroundColor: theme.palette.background.default,
+        '& .Mui-selected': {
+            color: theme.palette.primary.contrastText,
+            backgroundColor: theme.palette.primary.main,
+        },
+    },
+    tabPanel: {
+        marginTop: theme.spacing(3),
+    },
+    indicator: {
+        display: 'none',
+    },
+    applicationBox: {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        backgroundColor: getMaskColor(theme).twitterBackground,
+        borderRadius: '8px',
+        cursor: 'pointer',
+        height: 100,
+        '&:hover': {
+            transform: 'scale(1.05) translateY(-4px)',
+            boxShadow: theme.palette.mode === 'light' ? '0px 10px 16px rgba(0, 0, 0, 0.1)' : 'none',
+        },
+    },
+    applicationWrapper: {
+        marginTop: 4,
+        display: 'grid',
+        gridTemplateColumns: '123px 123px 123px 123px',
+        gridTemplateRows: '100px',
+        rowGap: 12,
+        justifyContent: 'space-between',
+        height: 324,
+    },
+    applicationImg: {
+        width: 36,
+        height: 36,
+        marginBottom: 10,
+    },
+    disabled: {
+        pointerEvents: 'none',
+        opacity: 0.5,
+    },
+}))
+
+export interface MaskAppEntry {
+    title: string
+    img: string
+    onClick: any
+    supportedChains?: ChainId[]
+    hidden: boolean
+    walletRequired: boolean
+}
+
+interface MaskApplicationBoxProps {
+    secondEntries?: MaskAppEntry[]
+    secondEntryChainTabs?: ChainId[]
+}
+
+export function MaskApplicationBox({ secondEntries, secondEntryChainTabs }: MaskApplicationBoxProps) {
+    const { classes } = useStyles()
+    const currentChainId = useChainId()
+    const account = useAccount()
+    const selectedWallet = useWallet()
+    //#region Encrypted message
+    const openEncryptedMessage = useCallback(
+        (id?: string) =>
+            MaskMessages.events.requestComposition.sendToLocal({
+                reason: 'timeline',
+                open: true,
+                options: {
+                    startupPlugin: id,
+                },
+            }),
+        [],
+    )
+    //#endregion
+
+    //#region Claim All ITO
+    const {
+        open: isClaimAllDialogOpen,
+        onOpen: onClaimAllDialogOpen,
+        onClose: onClaimAllDialogClose,
+    } = useControlledDialog()
+    //#endregion
+
+    //#region Swap
+    const { openDialog: openSwapDialog } = useRemoteControlledDialog(PluginTraderMessages.swapDialogUpdated)
```

Comment:
Why not to collect CompositionDialogEntry on plugin SNSAdaptor.

Reply:
Because application box is not generated from SNSAdaptor definition automatically.